### DISCUSSION TITLE
Statistics Reporting 0.13 Listeners per Hour When 1 Listener is Continuously Listening

### DISCUSSION THREAD

**Discussion by FendleTruck** [AzuraCast/AzuraCast#4737] [Q&A]
I had a server running for a while on a docker installation, and I recently moved to new hardware.

I backed up my AzuraCast configuration, and installed the Docker container from the repository on my new hardware, then restored my AzuraCast configuration.

Since doing so, there are some anomalies in the statistics. For example, it's not public yet, so I'm testing it by listening to it continuously and verifying the features work. In any given hour, "Listeners by Hour" shows 0.13 active listeners for every hour.

The listeners overview correctly shows the duration as several hours of listening.

"Listeners by Day" is just blank, as is the "Listeners per Station" graph.

I'm correctly reporting the number of active listeners per station on every page, but the "Listeners by Day", "Listeners per Station" are blank, and "Listeners by Hour" is showing 0.13 (7.7x too small?).

Originally I thought that the database wasn't accessible, however, it's clearly storing something. Next I thought that maybe I wasn't getting the number of listeners from IceCast, however, that information is available too.

Any ideas here?

**Comment by paulcjkp** [comment]
How long was your server active before you started listening?  It could possibly be averaging the listening hours from when the server went online, not just when someone started listening?  For example, if your server was up for 7 days before testing, and you listened on the 8th day, the average listening hours would be 0.125, rounded up to 0.13 (1 hour listened, over 8 days).

**Reply by FendleTruck** [reply]
Are these figures all averages since going online?

It's very possible it was online for 7 days before I began testing it.

Is "Listeners Per Station" some kind of average as well?

**Reply by paulcjkp** [reply]
I'm not really sure how it calculates, just guessing here.  Our station has 2 constant listeners (bots?), yet our listeners per hour during non-peak hours is between 1.04 and 1.16.  We did start off only having one constant listener, so perhaps that's why it's showing less than 2?  As for the blank items, it could just be that there's not enough data to generate a report yet?

**Reply by FendleTruck** [reply]
After having the server running for two months, I'm still seeing the same problem.

Listeners by hour has some other numbers, and looks reasonable considering this as an "all time" average.
"Listeners by Day" is always blank, no lines to be seen.

Is there any info out there to show how this is calculated?
I can't even tell what the X axis is supposed to be because there's no data on it at all, and the axis is unlabeled with no tick marks.
